nncf/quantization/quantizer_propagation.py

```diff
@@ -120,22 +124,21 @@ def __eq__(self, other):
     def __hash__(self):
         return hash(self.id)
 
-    @property
-    def potential_quant_configs(self) -> List[QuantizerConfig]:
-        return self._potential_quant_configs
 
 
 class TransitionStatus(Enum):
     SHOULD_TRANSITION = 0
     SHOULD_MERGE = 1
     SHOULD_NOT_TRANSITION = 2
+    SHOULD_WAIT_FOR_MERGE = 3
 
 
 class PropagationStrategy(Enum):
     CONSERVATIVE = 0  # While propagating up through a downward-branching node,
-                      # do not propagate if the propagation results in narrowing the list of
-                      # quantization variants available to quantizers on neighbouring branches
-    AGGRESSIVE = 1
+                      # do not merge at all, or ...
+    MODERATE = 1      # ... only merge for exact matches
+    AGGRESSIVE = 2    # ... merge common parts, and if a branch quantizer has options for scope narrowing in addition to
+                      # the common part, keep the quantizer on branch
 
```

Comment:
The `AGGRESSIVE` strategy will be used by default, and it is the one most likely to garner performance benefits.